### DISCUSSION TITLE
Cannot deploy emgucv on RedHat docker image due to "unable to load shared library"

### DISCUSSION THREAD

**Discussion by bogdan-todasca** [emgucv/emgucv#572] [General]
Note: it works fine when deploying onto a Ubuntu Docker image.

I am trying to include emgu.cv on a RedHat docker image but I cannot make it work. I get the generic error below which makes me think a dependency is missing. The RH image I am using is: `registry.access.redhat.com/ubi8/dotnet-50-runtime` and the emgucv version is **4.5.3.4721.**

`Unable to load shared library 'cvextern' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libcvextern: cannot open shared object file: No such file or directory
`
Is there a list of RH dependencies list that I need to add in the docker file or any other tips? Did anyone manage to run emgucv in a RH docker container?

**Comment by emgucv** [comment]
You will need a runtime package that support Redhat linux distribution. On nuget, for linux we only provide runtime package for debian distributions:
https://www.nuget.org/packages/Emgu.CV.runtime.ubuntu.20.04-x64/
https://www.nuget.org/packages/Emgu.CV.runtime.linux-arm/

Debian packages with native components are unlikely to be compatible with RedHat os. You will need to compile the runtime component from source code and package it up into a nuget file. At this time, we do not officially support RedHat, you will need to figure this out on your own, or use our consulting service under a contract ( https://www.emgu.com/wiki/index.php/Support_and_Services )
